### ISSUE TITLE
Testing Section

### ISSUE THREAD

**Issue by borispov** [flaviocopes/react-handbook#1]
Hello flaviocopes, really like your content, your blog. 

As for this react-handbook I'm sure I'll read it to refresh and improve my knowledge, 
And one thing that would be missing and very important is **Testing** section. 

I really like the code-splitting section, this is the first time I could read something that my brain can digest with examples. 

Maybe add content about Project Structure and best practice patterns.
and another thing that is related to code-splitting (maybe?) is separation of some Logic, for example if we have a Login component, you would create a separate file for handling the POST request. 

I'm sorry if this all sounds ridiculous, I'm relatively a beginner trying to take part in something and help :)

**Comment by flaviocopes** [comment]
Great ideas Boris, testing is definitely one thing I need to add, also project organization 👍

**Comment by hirako2000** [comment]
I agree with the testing section, throwing content about react is great, teaching that testing is upmost upmost important is even better.

I would personally even start with this: Working software isn't right if not tested. A book shall better start with testing, because many would read the get to action without bothering reading about testing (real story on Spring books).

TDD is not debatable these days, write test first, what testing framework work best with React, how redux reducer should be tested, pure function testing is easy, let's put it into action with full coverage test suite.

This free (open) book impresses me in many ways, I hope I could read something like this in 2015 when I discovered react, and  I do wish whatever else I was learning from put emphasis on testing.

If you need any contribution, give me a shout.
